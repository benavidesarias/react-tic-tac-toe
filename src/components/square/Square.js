import React from 'react';

function Square(props) {
    return (
        <button
            data-testid = 'square'
            className={props.className}
            onClick={props.onClick}
        >
            {props.value}
        </button>
    );
}

export default Square;