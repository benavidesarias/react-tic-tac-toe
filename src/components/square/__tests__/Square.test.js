import React from 'react';
import ReactDOM from 'react-dom';
import Square from '../Square';

import {render} from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

it("render", () => {
	const div = document.createElement("div");
	ReactDOM.render(<Square />,div);
});

it("renden with props.value X", () => {
	const {getByTestId} = render(<Square value='X' />);
	expect(getByTestId('square')).toHaveTextContent('X');
});

it("renden with props.value O", () => {
	const {getByTestId} = render(<Square value='O' />);
	expect(getByTestId('square')).toHaveTextContent('O');
});

it("snapshot", ()=>{
	const tree = renderer.create(<Square value='X' />).toJSON();
	expect(tree).toMatchSnapshot();
});