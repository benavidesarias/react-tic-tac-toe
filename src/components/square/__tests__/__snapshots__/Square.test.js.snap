// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<button
  data-testid="square"
>
  X
</button>
`;
